Assets/Mirror/Editor/MirrorAssetProcessor.cs

```diff
@@ -0,0 +1,129 @@
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using UnityEngine;
+using UnityEditor;
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace DigitalRuby.WeatherMaker
+{
+    /// <summary>
+    /// Pre-process define utility
+    /// </summary>
+    public class MirrorAssetPostprocessor : AssetPostprocessor
+    {
+        // add list of asset files to key off of along with pre-processor definition
+        // asset name / pre processor value key / value pair
+        internal static readonly KeyValuePair<string, string>[] preProcessors = new KeyValuePair<string, string>[]
+        {
+            new KeyValuePair<string, string>("/Mirror/Runtime/NetworkIdentity.cs", "MIRROR_NETWORKING_PRESENT"),
+        };
+
+        /// <summary>
+        /// Update pre-processor
+        /// </summary>
+        /// <param name="hasAsset">Whether the asset exists, if it does pre-processor is added, else it is removed if it exists</param>
+        /// <param name="preProcessor">The pre-processor to add or remove</param>
+        internal static void UpdatePreProcessor(bool hasAsset, string preProcessor)
+        {
+            string currBuildSettings = UnityEditor.PlayerSettings.GetScriptingDefineSymbolsForGroup(UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup);
+            string currBuildSettingsCompare = currBuildSettings;
+            if (hasAsset)
+            {
+                if (!currBuildSettings.Contains(preProcessor))
+                {
+                    currBuildSettings += ";" + preProcessor;
+                }
+            }
+            else
+            {
+                currBuildSettings = System.Text.RegularExpressions.Regex.Replace(currBuildSettings, ";?" + preProcessor + ";?", string.Empty);
+            }
+            if (currBuildSettings != currBuildSettingsCompare)
+            {
+                Debug.LogWarning("Updating preprocessor to " + currBuildSettings);
+                UnityEditor.PlayerSettings.SetScriptingDefineSymbolsForGroup(UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup, currBuildSettings);
+            }
+        }
+
+        private static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
+        {
+            foreach (string assetName in importedAssets)
```

Comment:
seem to be repeating this 3 times.  Don't repeat yourself   refactor into a method and call it from the 3 different OnXXX

Reply:
Ok